vm/rust/src/jsonrpc.rs

```diff
@@ -32,7 +39,35 @@ pub struct TransactionTrace {
     constructor_invocation: Option<FunctionInvocation>,
     #[serde(skip_serializing_if = "Option::is_none")]
     function_invocation: Option<FunctionInvocation>,
-    r#type: TransactionType,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    r#type: Option<TransactionType>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    state_diff: Option<ThinStateDiff>,
+}
+
+impl TransactionTrace {
+    pub fn make_legacy(&mut self) {
+        self.state_diff = None;
+        self.r#type = None;
+        if self.validate_invocation.is_some() {
+            self.validate_invocation.as_mut().unwrap().make_legacy()
+        }
+        if self.execute_invocation.is_some() {
+            match self.execute_invocation.as_mut().unwrap() {
+                ExecuteInvocation::Ok(fn_invocation) => { fn_invocation.make_legacy() }
+                _ => {}
+            }
+        }
+        if self.fee_transfer_invocation.is_some() {
+            self.fee_transfer_invocation.as_mut().unwrap().make_legacy()
+        }
+        if self.constructor_invocation.is_some() {
+            self.constructor_invocation.as_mut().unwrap().make_legacy()
+        }
+        if self.function_invocation.is_some() {
```

Comment:
common rust idiom is to use if let in such cases i.e. 
```
if let Some(actualValue) = self.function_invocation {
   // do something with
}
```
but it's up to you

Reply:
I knew there was a better way of doing this:)